src/server/tiered_storage.cc

```diff
@@ -268,38 +361,65 @@ void TieredStorage::Close() {
   op_manager_->Close();
 }
 
+void TieredStorage::SetMemoryLowLimit(size_t mem_limit) {
+  op_manager_->memory_low_limit_ = mem_limit;
+  VLOG(1) << "Memory low limit is " << mem_limit;
+}
+
 util::fb2::Future<string> TieredStorage::Read(DbIndex dbid, string_view key,
                                               const PrimeValue& value) {
   DCHECK(value.IsExternal());
   util::fb2::Future<string> future;
+  if (value.IsCool()) {
```

Comment:
I think we can pay the prive here and call TieredStorage::Read from below by contructing an std::function